sklearn/feature_extraction/text.py

```diff
@@ -1295,9 +1308,9 @@ class TfidfVectorizer(CountVectorizer):
     vocabulary_ : dict
         A mapping of terms to feature indices.
 
-    idf_ : array, shape = [n_features], or None
+    idf_ : array, shape = [n_features], or not defined
```

Comment:
"not defined" is not a python object, so better to just leave it out here. It's sufficiently documented in the docsting below. 

Also,
  * please remove "=" after "shape" and make it a tuple while we are at it (cf. e.g. https://github.com/scikit-learn/scikit-learn/blob/a24c8b46/sklearn/linear_model/logistic.py#L1113)
  * https://github.com/scikit-learn/scikit-learn/pull/10899#discussion_r178742440 still needs addressing


The same applies to the TfidfTransformer docstring. Thanks.

Reply:
It seems that throughout the sklearn code base None is used when a variable is not set to any value. Should the idf_ be set to None when use_idf=False

Note, I don't have much python experience, and I am still learning it. Thanks for the comments.

Reply:
`None` is frequently used in scikit-learn as a default value for parameters, but here this attribute will not return None, it will raise an `AttributeError` instead when use_idf=False